featurizers/fingerprints.py

```diff
@@ -4,65 +4,138 @@
 import numpy as np
 import pandas as pd
 import rdkit.Chem as Chem
-from rdkit.Chem.rdMolDescriptors import GetMorganFingerprintAsBitVect
 
 from base import FingerprintTransformer
 
+"""
+If during multiprocessing occurs MaybeEncodingError, first check if there isn't thrown any exception inside
+worker function! (That error isn't very informative and this tip might save you a lot of time)
+"""
 
-class MorganFingerprintAsBitVect(FingerprintTransformer):
+"""
+fp_descriptors needs to be here or inside _transform() of a specific class (cannot be defined inside the __init__() of
+that class), otherwise pickle gets angry:
+        TypeError: cannot pickle 'Boost.Python.function' object
+"""
+
+
+class MorganFingerprint(FingerprintTransformer):
     def __init__(
         self,
         radius: int = 2,
         n_bits: int = 2048,
         use_chirality: bool = False,
         use_bond_types: bool = True,
         use_features: bool = False,
+        result_type: str = "default",
+        n_jobs: int = 1,
+    ):
+        assert result_type in ["default", "as_bit_vect", "hashed"]
+
+        super().__init__(n_jobs)
+
+        self.fp_args = {
+            "radius": radius,
+            "useChirality": use_chirality,
+            "useBondTypes": use_bond_types,
+            "useFeatures": use_features,
+        }
```

Comment:
To muszą być osobne atrybuty, inaczej to nie zadziała z grid searchem. Polecam zapoznać się z oficjalnym tutorialem do customowych estymatorów w Scikit-learn: https://scikit-learn.org/stable/developers/develop.html. W szczególności na przykład:

> In addition, **every keyword argument accepted by __init__ should correspond to an attribute on the instance**. Scikit-learn relies on this to find the relevant attributes to set on an estimator when doing model selection.